vpnmanager/vpnservicemanager/vpnprotocol/openvpn/src/test/java/com/kape/openvpn/domain/controllers/StopProcessControllerTest.kt

```diff
@@ -163,4 +178,33 @@ internal class StopProcessControllerTest {
         // then
         assert(result.isSuccess)
     }
+
+    @Test
+    fun `must invoke hold release job cancel when stopping`() = runBlocking {
+        // given
+        val context: Context = ApplicationProvider.getApplicationContext()
+        val jobMock: IJob = JobMock()
+        val cancelHoldReleaseJobMock: ICancelHoldReleaseJob = GivenUsecase.cancelHoldReleaseJob(
+            job = jobMock
+        )
+        val isProcessRunningMock: IIsProcessRunning = IsProcessRunningMock(succeed = true)
+        val closeSocketMock: ICloseSocket = CloseSocketMock(succeed = true)
+        val stopProcessMock: IStopProcess = StopProcessMock(succeed = true)
+        val clearCache: IClearCache = ClearCacheMock(succeed = true)
+        val stopProcessController: IStopProcessController = GivenController.stopProcessController(
+            context = context,
+            cancelHoldReleaseJob = cancelHoldReleaseJobMock,
+            isProcessRunning = isProcessRunningMock,
+            closeSocket = closeSocketMock,
+            stopProcess = stopProcessMock,
+            clearCache = clearCache
+        )
+
+        // when
+        val result = stopProcessController()
+
+        // then
+        assert(result.isSuccess)
+        assert((jobMock as JobMock).invocationsPerformed[JobMock.MethodSignature.CANCEL_JOB] == 1)
```

Comment:
To avoid the cast here, we can just remove the type declaration as IJob in line 186.

Reply:
it is intentional. I want the `When` of the test to operate using the official `IJob` interface and what it offers.